### ISSUE TITLE
Increase 'live search' delay?

### ISSUE THREAD

**Issue by gavanderhoorn** [rosindex/rosindex.github.io#110]
On my system at least, using the [live search](http://rosindex.github.io/search/) seems to filter too fast. With some queries (single letters fi), this makes my browser appear to hang. Would it be an idea to increase the delay/timeout between input -> search?

I don't know how you implemented the search functionality, but using the built-in Firefox developers tools it seems `lunr.js` is (ultimately) used? I couldn't find out whether the delay is actually configurable, but the same developer tools lead me to (in `jquery.lunr.search.js`):
```javascript
  // prevent search from running too often
  var debounce = function(fn) {
    var timeout;
    var slice = Array.prototype.slice;

    return function() {
      var args = slice.call(arguments),
          ctx = this;

      clearTimeout(timeout);

      timeout = setTimeout(function () {
        fn.apply(ctx, args);
      }, 100);
    };
  };
```

Perhaps increase that `100` there?

**Comment by jbohren** [comment]
Yeah, I've been holding off on trying to optimize the search to see how well it works in practice. This seems like a good idea though. I'm pretty sure that `100` is what would need to be increased.

I'm also considering adding an additional search mode. The current one returns each *version* of a package as a separate result, but it might be just as helpful to just show a single result listing all versions of a package from a given repository. This would speed up the search results dramatically.

**Comment by gavanderhoorn** [comment]
> I'm also considering adding an additional search mode. The current one returns each version of a package as a separate result, but it might be just as helpful to just show a single result listing all versions of a package from a given repository. This would speed up the search results dramatically.

Hm, I 'always' search for 'pkg_name ros_distro'. That actually works well enough. But immediately reducing the search space would probably make things even faster.
